README.md

```diff
@@ -1,58 +1,106 @@
-
-[![js-standard-style](https://img.shields.io/badge/code%20style-standard-brightgreen.svg?style=flat)](https://github.com/feross/standard)
- 
-## Screenshots
-Include logo/demo screenshot etc.
-
-## Tech/framework used
-Ex. -
-
-<b>Built with</b>
-- [Electron](https://electron.atom.io)
+Music Library - exercise for SQL, databases and **CRUD** operations.
 
-## Features
-What makes your project stand out?
+## Motivation
 
-## Code Example
-Show what the library does as concisely as possible, developers should be able to figure out **how** your project solves their problem by looking at the code example. Make sure the API you are showing off is obvious, and that your code is short and concise.
+- Learn how to design test for each operation.
+- Learn how to route user request to handler.
+- Learn how to handle error of user request.
+
+## What will learn from this exercise
+
+### API flow of HTML request:
+
+| API flow           | Purpose                                   |
+| ------------------ | ----------------------------------------- |
+| index.js           | process config env                        |
+| src/app.js         | process routers                           |
+| src/routes/\*      | process get/post/put/patch controllers    |
+| src/controllers/\* | process back-end controllers and response |
+
+### Database setup for each test:
+
+To ensure tests have a fresh database each time they run.
+
+```
+// package.json
+ ...
+ "scripts": {
+    "migrate": "node scripts/migrate.js",
+    "prestart": "node scripts/create-database.js && npm run migrate",
+    "start": "nodemon -r dotenv/config index.js",
+    "pretest": "node scripts/create-database.js test && npm run migrate test",
+    "test": "mocha tests/**/*.js --exit --recursive --timeout 60000 --file ./scripts/test-setup.js",
+    "posttest": "node scripts/drop-database.js test"
+  },
+  ...
+```
+
+### A top-level directory layout
+
+    .
+    ├── migrations         # Setup databases before each test
+    ├── scripts            # Scripts for migrations
+    ├── src
+    │      ├── routes      # Source of routers
+    │      ├── controllers # Source of request handlers
+    │      └── db          # define query of pg
+    ├── tests              # test files for requests
+    └── README.md
 
 ## Installation
-Provide step by step series of examples and explanations about how to get a development env running.
+
+```
+$ git clone https://github.com/DavidCheungTW/music-library.git
+$ cd music-library
+$ npm install
+```
 
 ## API Reference
 
-Depending on the size of the project, if it is small and simple enough the reference docs can be added to the README. For medium size to larger projects it is important to at least provide a link to where the API reference docs live.
+```
+// package.json
+...
+  "devDependencies": {
+    "chai": "^4.3.7",
+    "dotenv": "^16.0.3",
+    "mocha": "^10.2.0",
+    "nodemon": "^2.0.20",
+    "supertest": "^6.3.3"
+  },
+  ...
+```
```

Comment:
As above dropping a code block like this isnt the right way to go about this. YOu could just just a simple bullet point list:
```
Development dependencies:
- chai
- dotenv
- mocha
- nodemon
- supertest
```